### DISCUSSION TITLE
select multiple colors invert and fill

### DISCUSSION THREAD

**Discussion by Litnsun** [ImageMagick/ImageMagick#5053] [Help] [answered]
![image](https://user-images.githubusercontent.com/103970051/163891374-392907f3-c5bd-4452-8df7-c0a8e18bd13d.png)
Hello, I'm new to imagemagick, I don't know if this is the right place to ask, since the forums seem to be archived.
I'm using ImageMagick-7.1.0-Q16 in Windows 10 Pro 64b.
I want to select multiple colors with -fuzz 20%, expand the selection by 1 or 2 pixels, invert the selection and then fill it with black.
let's say in this case the colors are `#d4c491
#ffff7f
#ff6969`
The idea is to remove the background so that ocr performs better.
The code I have as of now is this, it fills the colors with white and then fills non-white to black (since there is no shading, I'm not sure if it's ideal for ocr).
`magick convert test.png -fuzz 20% -fill white -opaque "#d4c491" -fuzz 20% -fill white -opaque "#ffff7f" -fuzz 20% -fill white -opaque "#ff6969" -fill black +opaque "#ffffff" result.png`
what I want to achieve is something like this
![image](https://user-images.githubusercontent.com/103970051/163893036-cd065772-7b42-42d1-a21b-cae9208de04e.png)

**Comment by snibgo** [comment, marked as answer]
Your result.png has a black background, with white letters. If your input is haaa.png, we can take the darker pixels of the two images:
```
magick haaa.png result.png -compose Darken -composite x.png
```
![x](https://user-images.githubusercontent.com/33812368/163897750-21691c22-aa59-4967-937b-87727766a8d8.png)


We can combine the two commands:
```
magick haaa.png ( +clone -fuzz 20% -fill white -opaque "#d4c491"
 -fuzz 20% -fill white -opaque "#ffff7f" -fuzz 20% -fill white -opaque "#ff6969"
 -fill black +opaque "#ffffff" ) -compose Darken -composite result2.png
```
If you use bash, escape the parentheses: \( and\ ).

![result2](https://user-images.githubusercontent.com/33812368/163897767-bf834009-20e3-4e3b-9d23-586351b96e46.png)

**Reply by Litnsun** [reply]
thank you, it works. I added this part
` -morphology dilate octagon`
so that the selection is expanded slightly, in case someone is looking for it.
`magick haaa.png ( +clone -morphology dilate octagon -fuzz 20% -fill white -opaque "#d4c491"  -fuzz 20% -fill white -opaque "#ffff7f" -fuzz 20% -fill white -opaque "#ff6969" -fill black +opaque "#ffffff" ) -compose Darken -composite result2.png`
